src/main/java/org/spongepowered/common/event/SpongeEventManager.java

```diff
@@ -211,12 +213,26 @@ public void registerListener(PluginContainer plugin, Object listenerObject) {
 
                     handlers.add(createRegistration(plugin, eventClass, listener, handler));
                 } else {
-                    SpongeImpl.getLogger().warn("The method {} on {} has @{} but has the wrong signature", method, handle.getName(),
-                            Listener.class.getName());
+                    invalidMethods.add(method);
                 }
             }
         }
 
+        // getMethods() doesn't return private methods. Do another check to warn
+        // about those.
+        for (Class<?> handleParent = handle; handleParent != Object.class; handleParent = handle.getSuperclass()) {
+            for (Method method : handleParent.getDeclaredMethods()) {
+                if (method.getAnnotation(Listener.class) != null && !isValidHandler(method)) {
+                    invalidMethods.add(method);
+                }
+            }
+        }
+
+        for (Method method : invalidMethods) {
+            SpongeImpl.getLogger().warn("The method {} on {} has @{} but has the wrong signature", method,
```

Comment:
Seeing as your changing this, it would probably be good to give a more helpful error message (eg. if its private warn specifically that event listeners cannot be private).